internal/controller/accessmanagement_controller.go

```diff
@@ -105,23 +110,38 @@ func (r *TemplateManagementReconciler) Reconcile(ctx context.Context, req ctrl.R
 					continue
 				}
 			}
+			for _, credentialName := range rule.Credentials {
+				keepCredentials[getNamespacedName(namespace, credentialName)] = true
+				if systemCredentials[credentialName] == nil {
+					errs = errors.Join(errs, fmt.Errorf("credential %s/%s is not found", r.SystemNamespace, credentialName))
+					continue
+				}
+				err = r.createCredential(ctx, namespace, credentialName, systemCredentials[credentialName])
+				if err != nil {
+					errs = errors.Join(errs, err)
+					continue
+				}
```

Comment:
nit: could be simplified to oneliner (two with the `\n`) (applicable for the loops above)

```go

errs = errors.Join(errs, r.createCredential(ctx, namespace, credentialName, systemCredentials[credentialName]))
```